### ISSUE TITLE
Roadmap v2

### ISSUE THREAD

**Issue by EmmanuelBeziat** [EmmanuelBeziat/emmanuelbeziat#10]
Here we go. It’s already time to move to VueJS 2.x so I can stick with up-to-dates versions of each plugins.

# Workflow

First things first, I want to have a nice workflow with this. Automated tasks, scripts, etc. So I wanna get the best of webpack before going on Core.

* [ ] Auto-optimize images
* [ ] Auto-build SVG sprite when dropping svg files in folder

**Comment by reverland** [comment]
hi, I inspect [Phenomic](https://phenomic.io/) which you introduced to me several months ago.

It provide a new idea on howto generate feed and sitemap. which I think is more neat. In that way, one can satisfy their needs more 'pluggable' with pure js, no more extra template needed.

I don't know if a meta json file for blogs is needed, for I use this file to provide navigation info for views like archieves. but one can also just append this info to webpack just in the build process.

**Comment by EmmanuelBeziat** [comment]
@reverland Hi!
I almost forgot about Phenomic, thanks for the reminder. I definitely have to check what they're doing with their webpack.
It appear that they also generate all the pages/articles directly in some html files : [check on putaindecode's github](https://github.com/putaindecode/putaindecode.io/tree/gh-pages/fr/articles/docker)

But if I'm totally for having a nicer way to generate stuff (like feed or rss), i'm not really a big fan of having a bunch of html files… So if possible, I’ll stick with a single page app.

I tried to check all the "server tips config" that I could find over the web, but none of it seems to work. So I’m gonna give a try at the webpack prerendering system. If there's still nothing, I’ll try deeper with the new vue 2 server side rendering, but right now I didn’t get it and how to use it…

**Comment by reverland** [comment]
yeah, phenomic generate html files before serve. If seo really important, generate posts' html may not be a good idea.(but if you don't use the strange 404 hack for github, it may not a problem?) google can fetch views and render it properly, but I haven't seen how share tools works.

ssr may be useful for seo, slow agent and agent with no js support, but you need a server which will render the blogs for you on the fly. you may refer to vue-hackernews-2.0 example for how ssr works.

I prepare to port phenomic plugins to use, some day...

**Comment by EmmanuelBeziat** [comment]
@reverland For the record, I just discovered https://nuxtjs.org/ a framework built on top of vue 2, using SSR and precised to working great as a static-site generator… I’ll give it a try.

**Comment by reverland** [comment]
its for vue ssr(mean you need a server)...if you like, is a good choice!
